pkg/cloud/cloud.go

```diff
@@ -0,0 +1,31 @@
+package cloud
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	awsPlatformType = "aws"
+)
+
+// GetKubeletHostnameOverride returns correct hostname for kubelet if it should
+// be overridden, or an empty string otherwise.
+func GetKubeletHostnameOverride(platformType string) (string, error) {
+	var hostname string
+	var err error
+	platformType = strings.ToLower(platformType)
+	switch platformType {
+	case awsPlatformType:
+		hostname, err = getAWSMetadataHostname()
+		if err != nil {
+			return "", err
+		}
```

Comment:
Why not just return here, then you don't need the final return or to have the vars on 15 and 16. I think this would make it cleaner
```suggestion
		return getAWSMetadataHostname()
```